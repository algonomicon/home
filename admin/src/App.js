import { Component } from 'inferno'
import { BrowserRouter } from 'inferno-router'
import { Provider as StoreProvider } from 'inferno-mobx'
import { Provider as RenderProvider } from 'inferno-fela'
import { ThemeProvider } from 'inferno-fela'
import { createRenderer } from 'fela'
import Routes from './Config/Routes'
import Stores from './Config/Stores'
import GlobalStyles from './Static/Assets/GlobalStyles.css'

const renderer = createRenderer()
renderer.renderStatic(GlobalStyles.toString())

const theme = {
  hufflepuff: '#9370DB',
  ravenclaw: '#5F527A',
  gryffindor: '#FDCA7A'
}

class App extends Component {
  render() {
    return (
      <RenderProvider renderer={renderer}>
        <ThemeProvider theme={theme}>
          <StoreProvider {...Stores}>
            <BrowserRouter>
              {Routes}
            </BrowserRouter>
          </StoreProvider>
        </ThemeProvider> 
      </RenderProvider>
    )
  }
}

export default App