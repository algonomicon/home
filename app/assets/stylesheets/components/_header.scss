header {
  background:url(/assets/nebula.jpg) no-repeat center;
  background-size:cover;
  height:35vh;
  transition:all 1s ease;

  .wrapper {
    position:relative;
    height:100%;

    .masthead {
      position:relative;
      margin-top:10px;
      display:flex;
      align-items:center;
      justify-content:space-between;

      .logo {
        a {
          display:block;

          svg {
            fill:white;
            max-height:25px;
            width:auto;
            shape-rendering: crispedges;
            transition:all 0.3s ease;
          }

          &:hover {
            svg {
              opacity:1;
            }
          }
        }
      }
      
      nav {
        display:flex;
        align-items:center;

        a {
          color:white;
          margin:0 10px;

          &:hover {
            color:$rust1;
          }
        }
        span {
          color:white;
          margin:0 5px;
          font-weight:bold;
          cursor:default;
        }
      }
    }

    .intro {
      position:absolute;
      width:100%;
      left:10px; bottom:20px;

      .tagline {
        color:white;
        @include font-size(14);

      }
      .title {
        color:white;
        margin-bottom:10px;
      }
      .search {
        display:flex;

        .input-group {
          position:relative;
          width:100%; max-width:400px;

          .input-icon {
            position:absolute;
            left:15px; top:50%;
            transform:translateY(-50%);

            i {
              @include font-size(20);
              color:$light3;
            }
          }

          input {
            width:100%;
            padding:20px;
            padding-left:50px;
            border-radius:3px;
          }
        }
        .search-button {
          display:block;
          color:white;
          background:linear-gradient(to left, $rust1, $rust2);
          padding:20px 40px;
          border-radius:3px;
          margin:0 10px;
        }
      }
    }
  }

  &.grayscale {
    filter:grayscale(1);
    transition:all 3s ease;
  }
}