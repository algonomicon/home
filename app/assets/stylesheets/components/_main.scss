body {
  transition:all 1s ease;
  
  &.grayscale {
    filter:grayscale(1);
    transition:all 3s ease;
  }

  main {
    min-height:60vh;
    padding-bottom:40px;
  }
}