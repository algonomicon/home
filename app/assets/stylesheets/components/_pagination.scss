.pagination {
  padding:5px 0;
  margin-left:-5px;
  
  span {
    a {
      color:$light3;
      padding:5px;

      &:hover {
        color:$purple2;
      }
    }
    &.current {
      padding:5px;
      color:$dark1;
      font-weight:bold;
    }
  }
}