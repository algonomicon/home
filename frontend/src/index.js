import { render } from 'inferno'
import { BrowserRouter } from 'inferno-router'
import { Provider as StoreProvider } from 'inferno-mobx'
import { Provider as RenderProvider } from 'inferno-fela'
import { createRenderer } from 'fela'
import GlobalStyles from './Static/Assets/GlobalStyles.css'
import Routes from './Config/Routes'
import Stores from './Config/Stores'
import './Static/Assets/AppStyles.scss'
import * as serviceWorker from './serviceWorker'


const renderer = createRenderer()
renderer.renderStatic(GlobalStyles.toString())

render((
  <RenderProvider renderer={renderer}>
    <StoreProvider {...Stores}>
      <BrowserRouter>
        {Routes}
      </BrowserRouter>
    </StoreProvider>
  </RenderProvider>
), document.getElementById('root'))

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister()
