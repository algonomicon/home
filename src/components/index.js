export { default as Categories } from "./categories"
export { default as Feed, FeedQuery } from "./feed"
export { default as Footer } from "./footer"
export { default as Header } from "./header"
export { default as Layout } from "./layout"
export { default as Logo } from "./logo"
export { default as Main } from "./main"
export { default as Minion } from "./minion"
export { default as Search } from "./search"
export { default as Sidebar } from "./sidebar"
export { default as Sort } from "./sort"
export { default as Tags } from "./tags"
export { default as Wrapper } from "./wrapper"
