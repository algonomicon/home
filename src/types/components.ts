export type FeedProps = {
  data: {
    allMarkdownRemark: {
      nodes: {
        excerpt: string
        frontmatter: {
          title: string
          slug: string
          date: string
        }
        fields: {
          collection: string
        }
      }[]
    }
  }
}

export type FeedItemProps = {
  data: {
    excerpt: string
    frontmatter: {
      title: string
      slug: string
      date: string
    }
    fields: {
      collection: string
    }
  }
}
