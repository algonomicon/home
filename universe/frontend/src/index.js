import { render } from 'inferno'
import { createRenderer } from 'fela'
import { Provider } from 'inferno-fela'
import App from './App/App'
import GlobalStyles from './App/GlobalStyles.css'
import * as serviceWorker from './serviceWorker'

const renderer = createRenderer()
renderer.renderStatic(GlobalStyles.toString())

render((
  <Provider renderer={renderer}>
    <App />
  </Provider>
), document.getElementById('root'))

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister()
