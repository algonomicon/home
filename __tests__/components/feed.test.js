import React from "react"
import renderer from "react-test-renderer"

import { Feed } from "../../src/components"

describe("Feed", () => {
  it("renders correctly", () => {
    const data = {
      articles: {
        edges: [
          {
            node: {
              _type: "article",
              _createdAt: "2019-05-02T03:13:19Z",
              title: "Getting Flux and Cuda to Work with Julia Pro",
              description:
                "The fundamental concepts of classical physics, space, time, mass, and derived concepts, velocity, momentum, force, angular momentum, energy ... all rest on the principle that material points have trajectories. They are defined as lines in space-time. Even the dynamics of continuous, solid or fluid media describes the trajectories of the material points which constitute the bodies in motion. But the indeterminacy relation of Heisenberg prevents quantum particles from having such classical trajectories, since their position and velocity can not be exactly defined at the same time. How then can it explain all the appearances which legitimize the fundamental concepts of classical physics?",
              slug: {
                current: "getting-flux-and-cuda-to-work-with-julia-pro",
              },
            },
          },
        ],
      },
      datasets: {
        edges: [
          {
            node: {
              _type: "dataset",
              _createdAt: "2019-04-17T17:03:27Z",
              title: "MNIST",
              description: "Handwritten digits",
              slug: {
                current: "mnist",
              },
            },
          },
        ],
      },
      papers: {
        edges: [
          {
            node: {
              _type: "paper",
              _createdAt: "2019-04-18T15:33:31Z",
              title: "A Neural Algorithm of Artistic Style",
              abstract:
                "In fine art, especially painting, humans have mastered the skill to create unique visual experiences through composing a complex interplay between the content and style of an image. Thus far the algorithmic basis of this process is unknown and there exists no artificial system with similar capabilities. However, in other key areas of visual perception such as object and face recognition near-human performance was recently demonstrated by a class of biologically inspired vision models called Deep Neural Networks. Here we introduce an artificial system based on a Deep Neural Network that creates artistic images of high perceptual quality. The system uses neural representations to separate and recombine content and style of arbitrary images, providing a neural algorithm for the creation of artistic images. Moreover, in light of the striking similarities between performance-optimised artificial neural networks and biological vision, our work offers a path forward to an algorithmic understanding of how humans create and perceive artistic imagery.",
              slug: {
                current: "a-neural-algorithm-of-artistic-style",
              },
            },
          },
        ],
      },
      projects: {
        edges: [
          {
            node: {
              _type: "project",
              _createdAt: "2019-04-18T16:24:58Z",
              title: "Coming Soon",
              description:
                "Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approaches to corporate strategy foster collaborative thinking to further the overall value proposition. Organically grow the holistic world view of disruptive innovation via workplace diversity and empowerment.",
              slug: {
                current: "coming-soon",
              },
            },
          },
        ],
      },
      snippets: {
        edges: [
          {
            node: {
              _type: "snippet",
              _createdAt: "2019-05-04T23:31:11Z",
              title: "Hello World",
              description: "Lorem ipsum dolor sit amet.",
              slug: {
                current: "hello-world",
              },
            },
          },
        ],
      },
    }

    const tree = renderer.create(<Feed data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
