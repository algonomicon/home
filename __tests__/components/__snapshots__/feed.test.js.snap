// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<div>
  <h6
    className="sc-htpNat iacXlz"
  >
    Feed
  </h6>
  <div
    className="sc-bxivhb kphxJa"
  >
    <div>
      <a
        className="sc-ifAKCX gYXmWr"
        href="/snippets/hello-world"
      >
        <h4
          className="sc-EHOje eYGxhL"
        >
          Hello World
        </h4>
        <p
          className="sc-bZQynM hDSPrr"
        >
          Lorem ipsum dolor sit amet.
          ...
        </p>
        <small
          className="sc-gzVnrw bpUwpJ"
        >
          Snippet from 
          6 days ago
        </small>
      </a>
    </div>
    <div>
      <a
        className="sc-ifAKCX gYXmWr"
        href="/articles/getting-flux-and-cuda-to-work-with-julia-pro"
      >
        <h4
          className="sc-EHOje eYGxhL"
        >
          Getting Flux and Cuda to Work with Julia Pro
        </h4>
        <p
          className="sc-bZQynM hDSPrr"
        >
          The fundamental concepts of classical physics, space, time, mass, and derived concepts, velocity, mo
          ...
        </p>
        <small
          className="sc-gzVnrw bpUwpJ"
        >
          Article from 
          9 days ago
        </small>
      </a>
    </div>
    <div>
      <a
        className="sc-ifAKCX gYXmWr"
        href="/projects/coming-soon"
      >
        <h4
          className="sc-EHOje eYGxhL"
        >
          Coming Soon
        </h4>
        <p
          className="sc-bZQynM hDSPrr"
        >
          Leverage agile frameworks to provide a robust synopsis for high level overviews. Iterative approache
          ...
        </p>
        <small
          className="sc-gzVnrw bpUwpJ"
        >
          Project from 
          22 days ago
        </small>
      </a>
    </div>
    <div>
      <a
        className="sc-ifAKCX gYXmWr"
        href="/papers/a-neural-algorithm-of-artistic-style"
      >
        <h4
          className="sc-EHOje eYGxhL"
        >
          A Neural Algorithm of Artistic Style
        </h4>
        <p
          className="sc-bZQynM hDSPrr"
        >
          In fine art, especially painting, humans have mastered the skill to create unique visual experiences
          ...
        </p>
        <small
          className="sc-gzVnrw bpUwpJ"
        >
          Paper from 
          22 days ago
        </small>
      </a>
    </div>
    <div>
      <a
        className="sc-ifAKCX gYXmWr"
        href="/datasets/mnist"
      >
        <h4
          className="sc-EHOje eYGxhL"
        >
          MNIST
        </h4>
        <p
          className="sc-bZQynM hDSPrr"
        >
          Handwritten digits
          ...
        </p>
        <small
          className="sc-gzVnrw bpUwpJ"
        >
          Dataset from 
          23 days ago
        </small>
      </a>
    </div>
  </div>
</div>
`;
